app/tests/integration/test_openstack_builder.py

```diff
@@ -85,49 +101,65 @@ def test_initialize(
 
 
 @pytest.fixture(scope="module", name="image_ids")
-def image_ids_fixture(
+# overhead (e.g. to create new fixture) to reduce the amount of arguments seems to be too high.
+def image_ids_fixture(  # pylint: disable=too-many-positional-arguments,too-many-arguments
     image_config: types.ImageConfig,
     openstack_metadata: types.OpenstackMeta,
     test_id: str,
     proxy: types.ProxyConfig,
-    dockerhub_mirror: urllib.parse.ParseResult | None,
-):
+    callback_script: Path,
+    dockerhub_mirror_url: str,
+) -> list[str]:
     """A CLI run.
 
     This fixture assumes pipx is installed in the system and the github-runner-image-builder has
     been installed using pipx. See testenv:integration section of tox.ini.
     """
-    image_ids = openstack_builder.run(
-        cloud_config=openstack_builder.CloudConfig(
-            cloud_name=openstack_metadata.cloud_name,
-            dockerhub_cache=dockerhub_mirror,
-            flavor=openstack_metadata.flavor,
-            network=openstack_metadata.network,
-            proxy=proxy.http,
-            prefix=test_id,
-            upload_cloud_names=[openstack_metadata.cloud_name],
-        ),
-        image_config=config.ImageConfig(
-            arch=image_config.arch,
-            base=config.BaseImage.from_str(image_config.image),
-            microk8s="",  # "1.31-strict/stable", microk8s support will be removed
-            runner_version="",
-            name=f"{test_id}-image-builder-test",
-            juju="",  # "3.1/stable", juju support will be removed
-            script_config=config.ScriptConfig(
-                script_url=urllib.parse.urlparse(
-                    "https://raw.githubusercontent.com/canonical/github-runner-image-builder/"
-                    "eb0ca315bf8c7aa732b811120cbabca4b8d16216/tests/integration/testdata/"
-                    "test_script.sh"
-                ),
-                script_secrets={
-                    "TEST_SECRET": "SHOULD_EXIST",
-                    "TEST_NON_SECRET": "SHOULD_NOT_EXIST",
-                },
-            ),
-        ),
-        keep_revisions=1,
+    openstack_image_name = f"{test_id}-image-builder-test"
+    cli_args = [
+        "/usr/bin/sudo",
+        Path.home() / ".local/bin/github-runner-image-builder",
+        "run",
+        openstack_metadata.cloud_name,
+        openstack_image_name,
+        "--base-image",
+        image_config.image,
+        "--keep-revisions",
+        "1",
+        "--arch",
+        image_config.arch.value,
+        "--callback-script",
+        str(callback_script.absolute()),
```

Comment:
I need to use the cli as the callback functionality is currently only implemented in the cli.

Reply:
Its not, I have removed it.